lib/services/cordova-migration-service.ts

```diff
@@ -134,9 +161,133 @@ export class CordovaMigrationService implements ICordovaMigrationService {
+				}
+			}
+
+			let versionDisplayName = this.getDisplayNameForVersion(newVersion).wait();
+			this.$logger.info("Migrating to Cordova version %s", versionDisplayName);
+			let oldVersion = this.$project.projectData.FrameworkVersion;
+
+			_.each(this.$project.configurations, (configuration: string) => {
+				let oldPluginsList = this.$project.getProperty("CorePlugins", configuration);
+				let newPluginsList = this.migratePlugins(oldPluginsList, oldVersion, newVersion).wait();
+				this.$logger.trace("Migrated core plugins to: ", helpers.formatListOfNames(newPluginsList, "and"));
+				this.$project.setProperty("CorePlugins", newPluginsList, configuration);
+			});
+
+			let backedUpFiles: string[] = [],
+				backupSuffix = ".backup";
+			try {
+				_.each(this.$mobileHelper.platformNames, (platform) => {
+					this.$logger.trace("Replacing cordova.js file for %s platform ", platform);
+					let cordovaJsFileName = path.join(this.$project.getProjectDir().wait(), util.format("cordova.%s.js", platform).toLowerCase());
+					let cordovaJsSourceFilePath = this.$resources.buildCordovaJsFilePath(newVersion, platform);
+					this.$fs.copyFile(cordovaJsFileName, cordovaJsFileName + backupSuffix).wait();
+					backedUpFiles.push(cordovaJsFileName);
+					this.$fs.copyFile(cordovaJsSourceFilePath, cordovaJsFileName).wait();
+				});
+			} catch(error) {
+				_.each(backedUpFiles, file => {
+					this.$logger.trace("Reverting %s", file);
+					this.$fs.copyFile(file + backupSuffix, file).wait();
+				});
+				this.$errors.failWithoutHelp(error.message);
+			}
+			finally {
+				_.each(backedUpFiles, file => {
+					this.$fs.deleteFile(file + backupSuffix).wait();
+				});
+			}
+
+			this.$logger.info("Successfully migrated to version %s", versionDisplayName);
+		}).future<void>()();
+	}
+
+	public getSupportedPlugins(): IFuture<string[]> {
+		return (() => {
+			let version: string;
+			if(this.$project.projectData) {
+				version = this.$project.projectData.FrameworkVersion;
+			} else {
+				let selectedFramework = _.last(_.select(this.getSupportedFrameworks().wait(), (sv: Server.FrameworkVersion) => sv.DisplayName.indexOf(this.$projectConstants.EXPERIMENTAL_TAG) === -1));
+				version = selectedFramework.Version;
+			}
+
+			return this.pluginsForVersion(version).wait();
+		}).future<string[]>()();
+	}
+
+	private getSupportedWPFrameworks(): IFuture<string[]>{
+		return ((): string[]=> {
+			let validValues: string[] = [];
+			let projectSchema = this.$project.getProjectSchema().wait();
+			if(projectSchema) {
+				validValues = this.$projectPropertiesService.getValidValuesForProperty(projectSchema["WPSdk"]).wait();
+			}
+
+			return validValues;
+		}).future<string[]>()();
+	}
+
 	private parseMscorlibVersion(json: any): string {
 		return [json._Major, json._Minor, json._Build].join('.');
 	}
+
+	private isPluginSupported(plugin: string, version: string, migrationVersion: string): boolean {
```

Comment:
Can you make this method public and add it to the interface. I would like to reuse it when adding new plugin to project and validate it is supported. Also I'm not sure if this is the appropriate place for this method. I think it should be part of pluginsService.